File: backend/Responses/Auth/SignInResponse.cs
using backend.Models;

namespace backend.Responses.Auth;

public class SignInResponse : BaseResponse
{
    public string Token { get; }
    public UserWithoutPassword User { get; }

    public SignInResponse(
        string message,
        string token,
        RequestVariant variant,
        UserWithoutPassword user) : base(message, variant)
    {
        this.Token = token;
        this.User = user;
    }

    public override object ToObject()
    {
        return new
        {
            message = base.Message,
            token = this.Token,
            variant = base.Variant.ToString(),
            user = this.User.ToObject()
        };
    }
}

File: web/src/responses/auth/sign-in-response.ts
import { UserModel } from "@/models";
import { BaseResponse, BaseResponseProps } from "../base-response";

export interface SignInResponseProps extends BaseResponseProps {
  token: string;
  user: UserModel;
}

export class SignInResponse extends BaseResponse {
  public token: string;
  public user: UserModel;

  constructor({ message, variant, token, user }: SignInResponseProps) {
    super({ message, variant });

    this.token = token;
    this.user = user;
  }
}